apps/music/js/metadata/flac.js

```diff
@@ -54,11 +88,9 @@ var FLACMetadata = (function() {
    */
   function findVorbisCommentBlock(blobview) {
```

Comment:
Since we're expecting to find multiple blocks now, I think we should switch to using a callback instead of a Promise. My general rule for Promises is to use them for something that's simply async, but if it's more like an event (e.g. it can happen multiple times), a callback is better. This would eliminate the need for us to call this function inside of the handler (see line 78).

Also, since this function is doing more than just finding the Vorbis Comment block, I think we should rename it to `getMetadataBlocks`.